### PR TITLE
Don't rely on external shasum program to calculate sum of the APK.

### PR DESCRIPTION
**Pull request by TheBosZ**
Replaces PR #120

Windows has no direct way to kill a process based on port. Uses netstats and loops through the results to find the correct process to kill.

Also modify Process.run for the server to runInShell if on Windows.

Style nits.

### CODE REVIEW

File: lib/src/device.dart [hunk 1]
```diff
@@ -824,6 +826,26 @@ class AndroidDevice extends Device {
...
+
+      //List entries from netstat is formatted like so
+      // TCP    192.168.2.11:50945     192.30.252.90:443      LISTENING     1304
+      //This regexp is to find process where the the port exactly matches
+      RegExp pattern = new RegExp(':$_serverPort[ ]+');
+
+      //Split the columns by 1 or more spaces
+      RegExp columnPattern = new RegExp('[ ]+');
+      processes.forEach((String process){
+        if(process.contains(pattern)) {
```
Comment (abarth) [inline lib/src/device.dart]: Sorry, the "if" should be left aligned with the p in processes but there should be a space between the "f" and the "(":

```
    RegExp columnPattern = new RegExp('[ ]+');
    processes.forEach((String process){
    if (process.contains(pattern)) {
```

File: lib/src/device.dart [hunk 2]
```diff
@@ -824,6 +826,26 @@ class AndroidDevice extends Device {
...
+      List<String> processes = runSync(['netstat.exe','-ano']).split("\r");
+
+      //List entries from netstat is formatted like so
+      // TCP    192.168.2.11:50945     192.30.252.90:443      LISTENING     1304
+      //This regexp is to find process where the the port exactly matches
+      RegExp pattern = new RegExp(':$_serverPort[ ]+');
+
+      //Split the columns by 1 or more spaces
+      RegExp columnPattern = new RegExp('[ ]+');
+      processes.forEach((String process){
```
Comment (abarth) [inline lib/src/device.dart]: There should be a space between the `)` and the `{`

File: lib/src/device.dart (hunk 1 above)
Comment (abarth) [inline lib/src/device.dart]: There should be a space between the `f` and the `(`

**Comment by abarth** [comment]
LGTM with style nits

**Comment by TheBosZ** [comment]
Okay, I fixed the nits, rebased and pushed.

File: lib/src/device.dart (hunk 2 above)
Comment (abarth) [inline lib/src/device.dart]: ^^^^ One more white space.

**Pull request merged**